lib/ansible/modules/windows/win_file.ps1

```diff
@@ -116,7 +116,20 @@ if (Test-Path $path) {
     }
 
     if ($state -eq "directory") {
-        New-Item -Path $path -ItemType Directory -WhatIf:$check_mode | Out-Null
+        if (-not $check_mode) {
```

Comment:
This is not needed because we already do `-WhatIf:$check_mode`.
And the failure paths should be tested in check-mode to.